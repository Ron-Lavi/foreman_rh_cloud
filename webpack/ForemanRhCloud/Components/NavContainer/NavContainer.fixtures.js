import React from 'react';
import { noop } from 'patternfly-react';

export const props = {
  items: [
    {
      icon: 'some-icon',
      name: 'some-name',
      component: () => <p>test</p>,
      onClick: noop,
    },
  ],
};
