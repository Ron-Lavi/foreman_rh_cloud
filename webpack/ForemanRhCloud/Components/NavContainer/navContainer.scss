.dashboard {
  .nav_item {
    font-size: 12px;
    text-align: center;
    width: 130px;
  }
}
