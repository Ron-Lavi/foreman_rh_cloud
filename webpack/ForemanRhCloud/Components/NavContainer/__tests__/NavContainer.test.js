import React from 'react';
import { mount, testComponentSnapshotsWithFixtures } from '@theforeman/test';
import NavContainer from '../NavContainer';
import { props } from '../NavContainer.fixtures';

const fixtures = {
  'render without Props': {},
  'render with Props': props,
};

describe('NavContainer', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(NavContainer, fixtures));

  it('click should call onTabClick prop', () => {
    const onTabClick = jest.fn();
    const modifiedProps = {
      ...props,
    };
    modifiedProps.items[0].onClick = onTabClick;
    const wrapper = mount(<NavContainer {...modifiedProps} />);
    wrapper.find('NavItem a').simulate('click');
    expect(onTabClick).toBeCalled();
  });
});
