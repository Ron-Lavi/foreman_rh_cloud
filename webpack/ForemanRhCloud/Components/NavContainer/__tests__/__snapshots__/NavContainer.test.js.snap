// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavContainer rendering render with Props 1`] = `
<ForwardRef
  defaultActiveKey={0}
  id="basic-tabs-pf"
>
  <div
    className="dashboard"
  >
    <Nav
      bsClass="nav nav-tabs nav-tabs-pf"
      justified={false}
      pullLeft={false}
      pullRight={false}
      stacked={false}
    >
      <NavItem
        active={false}
        className="nav_item"
        disabled={false}
        eventKey={0}
        key="0"
        onClick={[Function]}
      >
        <Icon
          name="some-icon"
          size="2x"
          type="fa"
        />
        <p>
          some-name
        </p>
      </NavItem>
    </Nav>
    <TabContent
      animation={true}
      bsClass="tab"
      componentClass="div"
      mountOnEnter={false}
      unmountOnExit={false}
    >
      <TabPane
        bsClass="tab-pane"
        eventKey={0}
        key="0"
      >
        <component />
      </TabPane>
    </TabContent>
    <FullScreenModal
      showFullScreen={false}
      terminalProps={Object {}}
      toggleFullScreen={[Function]}
    />
  </div>
</ForwardRef>
`;

exports[`NavContainer rendering render without Props 1`] = `
<ForwardRef
  defaultActiveKey={0}
  id="basic-tabs-pf"
>
  <div
    className="dashboard"
  >
    <Nav
      bsClass="nav nav-tabs nav-tabs-pf"
      justified={false}
      pullLeft={false}
      pullRight={false}
      stacked={false}
    />
    <TabContent
      animation={true}
      bsClass="tab"
      componentClass="div"
      mountOnEnter={false}
      unmountOnExit={false}
    />
    <FullScreenModal
      showFullScreen={false}
      terminalProps={Object {}}
      toggleFullScreen={[Function]}
    />
  </div>
</ForwardRef>
`;
